File: 02comm/02uart_read.js
var SerialPort = require('serialport');
var port = new SerialPort("/dev/serial0", {baudRate:9600}, function(err){
	if(err){
		console.log('err:' + err.message);
	}else{
		var line = '';
		port.on('data', function(data){
			if(data =='\r'){
				console.log('recv:' + line +'\r');	
				line = '';
			}else{
				line +=data;
			}
			
		});
	}
});

File: 02comm/02uart_read.py
import serial

port = serial.Serial("/dev/serial0", baudrate=9600, timeout=3.0)


def readline():
    line = ''
    while True:
        ch = port.read().decode()
        if ch =='\n' or ch == '\r' or ch == '':
            if len(line) > 0 :
                return line
        else:
            line +=ch
    
try:
    if port.isOpen() :
        print('serial is ready.')
        while True:
            print("recv:" + readline()) 
finally:
    port.close()